core/src/mongosqltranslate.rs

```diff
@@ -63,12 +63,12 @@ fn get_mock_library_path() -> PathBuf {
 // and is responsible for determining the library name and path.
 // The library name and path are determined based on the operating system and architecture.
 // It is stored in a static variable to ensure that it is only loaded once.
-pub fn load_mongosqltranslate_library() {
+pub fn load_mongosqltranslate_library(mock_library: bool) {
     INIT.call_once(|| {
-        let library_path = if cfg!(test) {
+        let library_path = if mock_library {
             get_mock_library_path()
         } else {
-            get_library_path(LIBRARY_NAME)
+            get_library_path()
```

Comment:
The problem is that we use `cargo test` for all of our tests, and there's other tests that need the real library such as the `bad_host()` test. On the other hand, only one test needs the mock library and this doesn't seem likely to change. I guess we could also make a feature flag for the one test that needs the mock library: `library_load_and_run_command_test()`.

Reply:
Actually, I wrote that wrong. Its more like we have to make sure the `bad_host()` test doesn't have a library available, but this will never work if it just defaults to the mock library.

Reply:
What I'm trying to get at is that the mock library isn't really a mock library that can be used in the absence of the real one. It's more like a dummy library that we can use to test for loading and calling `run_command`, but nothing else. So it's really only useful for that one test.

Reply:
If you need a different library then it would mean that this is an integration test. Their are already cluster type tests under integration tests and they seem well suited for testing cluster types.
I looked into the history of that test.
It used to be for testing that if the uri was not pointing to a mongodb host, then we would fail with "server selection timeout".
I think the issue is coming from the change in cluster_type.rs.
In `pub async fn determine_cluster_type(client: &Client) -> MongoClusterType {`, if we can't get build info back we assume it's because we don't know the type of cluster and we swallow the error.
```
    let cmd_res: Document = match db.run_command(build_info_cmd).await {
        Ok(res) => res,
        Err(e) => {
            log::error!("Failed to run buildInfo command: {:?}", e);
            return MongoClusterType::UnknownTarget;
        }
    };
```
We should return `Result<MongoClusterType>` and bubble up the error.
This would fix the test (you will need to revert the change made in PR #240 for the error message) and let you only use the mock library when running unit tests I believe.

Reply:
I think the issue is here:
```
        load_mongosqltranslate_library();

        let (is_libmongosqltranslate_compatible_with_driver_version, libmongosqltranslate_version) =
            if get_mongosqltranslate_library().is_some() {
                let libmongosqltranslate_version = Self::get_libmongosqltranslate_version()?;

                // This appends "|libmongosqltranslate+<version>" to the app_name
                user_options
                    .client_options
                    .app_name
                    .as_mut()
                    .ok_or(Error::EmptyAppName)?
                    .push_str(
                        &("|libmongosqltranslate+".to_owned() + &libmongosqltranslate_version),
                    );

                let compatibility = Self::is_libmongosqltranslate_compatible_with_driver_version()?;

                log::info!("libmongosqltranslate version: `{0}`. ODBC driver version: `{1}`. The ODBC driver and libmongosqltranslate library is compatible: `{2}`.", libmongosqltranslate_version, *DRIVER_METRICS_VERSION, compatibility);

                (Some(compatibility), Some(libmongosqltranslate_version))
            } else {
                (None, None)
            };
```
The mock library can't handle the runCommands that occur before we know the cluster type.